src/main/java/uk/gov/ons/ctp/response/action/service/ActionPlanJobService.java

```diff
@@ -60,87 +71,76 @@ public ActionPlanJob findActionPlanJob(final UUID actionPlanJobId) {
     return actionPlanJobRepo.findByActionPlanFK(actionPlan.getActionPlanPK());
   }
 
-  public List<ActionPlanJob> createAndExecuteAllActionPlanJobs() {
-    final List<ActionPlanJob> executedJobs = new ArrayList<>();
-    actionPlanRepo
-        .findAll()
-        .forEach(
-            actionPlan -> {
-              final Date lastExecutionTime =
-                  new Date(
-                      nowUTC().getTime() - appConfig.getPlanExecution().getDelayMilliSeconds());
-              if (actionPlan.getLastRunDateTime() == null
-                  || actionPlan.getLastRunDateTime().before(lastExecutionTime)) {
-                ActionPlanJob job =
-                    ActionPlanJob.builder()
-                        .actionPlanFK(actionPlan.getActionPlanPK())
-                        .createdBy(CREATED_BY_SYSTEM)
-                        .build();
-                job = createAndExecuteActionPlanJob(job);
-                if (job != null) {
-                  executedJobs.add(job);
-                }
-              } else {
-                log.with("action_plan_id", actionPlan.getId())
-                    .with("action_plan_pk", actionPlan.getActionPlanPK())
-                    .debug("Job for plan has been run since last wake up - skipping");
-              }
-            });
-    return executedJobs;
+  public void createAndExecuteAllActionPlanJobs() {
+    List<ActionPlan> actionPlans = actionPlanRepo.findAll();
+    actionPlans.forEach(
+        actionPlan -> {
+          if (shouldCreateAndExecuteActionPlanJob(actionPlan)) {
+            createAndExecuteActionPlanJob(actionPlan);
+          }
+        });
+  }
+
+  private boolean shouldCreateAndExecuteActionPlanJob(ActionPlan actionPlan) {
+    if (hasActionPlanBeenRunSinceLastSchedule(actionPlan)) {
```

Comment:
the method stink a bit, considering is checking conditions and does not change any status. The multiple if for logging debug informations are quite bad. What about check hasActionableCases || hasActionPlanBeenRunSinceLastSchedule as false condition otherwise return true ?